lib/dns.js

```diff
@@ -192,10 +195,10 @@ function onlookupservice(err, host, service) {
 // lookupService(address, port, callback)
 exports.lookupService = function(host, port, callback) {
   if (arguments.length !== 3) {
-    throw new Error('Invalid arguments');
+    throw new TypeError(`Expected 3 arguments, got ${arguments.length}`);
```

Comment:
Well, the type of `lookupService` is a function that takes three arguments and returns its results - if you pass an incorrect number of arguments to it then you're misusing the type.

For example - when you call `setTimeout` with 0 arguments in browsers you get a TypeError. When you call `Array#reduce` with 1 argument on an empty array you get a TypeError about the missing argument, and so on.

I don't feel very strongly about this - but the language and most APIs I know throw type errors in this case.